doc/diagnostics/AddingTracepoints.md

```diff
@@ -0,0 +1,417 @@
+
+`<l>` is the level of the tracepoint (0 through 9). High frequency tracepoints, known as hot tracepoints, are assigned higher level numbers.
+
+`<f>` is an internal flag (Y/N), no longer used.
+
+`<symbol>` is the internal symbolic name of the tracepoint.
+
+`<template>` is the template (in double quotation marks) that is used to format the entry.
+
+
+## Adding the first tracepoint to a dynamic module
+
+Ok, you want to enable a new module for trace. You need to set up this module as a trace module. Fortunately this isn't too hard, but there are several steps to follow.
+
+1. Determine a unique and representative name for your module
+2. Create a .tdf file for your module
+3. The build
+4. Include the generated .c file in your module's build (this usually involves an update to a module.xml or CMakeLists.txt file)
+5. Add dummy `ut_xxx.h` and `ut_xxx.c` source files to your module
+6. Include the generated header file in the source files that will be issuing the tracepoint
+7. Add a call to the startup of your module to register it with trace engine at runtime
+8. Optionally add a call to shutdown your module with trace when it unloads
+
+1) Create a module name
+
+You are largely on your own here, but search the source to make sure you choose a new one.
+
+2) Create a .tdf file for your module
+
+Start it with the key value pair `Executable=modName` where modName is what you came up with in 1).
+
+Add a clause that lets the trace engine know which .dat file your tracepoints are to be recorded in. For an OpenJ9 component, this will be:
+
+`DATFileName=J9TraceFormat.dat`
+
+Add any of the optional submodules you may wish your component to use:
+
+`Submodules=j9vmutil,j9util`
+
+This tells your module that it's initialization code needs to call the trace initialization function (`UT_MODULE_LOADED`) for these modules as well. This code is automatically generated. If you do not include the submodules (typically from the statically linked modules that your dynamic module uses) that you are using then the tracepoints and asserts in these modules will be disabled when your module calls them.
+
+3) The build
+
+The tracegen step of a build creates a .c file and a header file for your module, with names like `ut_xxx.c` and `ut_xxx.h` where `xxx` is your module name.
+
+4) Add the `ut_xxx.c` file to your modules makefiles
+
+The .c file contains control information and tables for your trace component, so needs to be built into an object to be linked into your module.
+
+5) Add a dummy `ut_xxx.h` file to your module
+
+The dummy file should contain the following:
+
+```
+/*
+ * This is a placeholder for an autogenerated file.
+ * The actual file will be produced when the following
+ * command is run as part of the build process:
+ *    gmake -f makefile.buildtools tracing
+ */
+
+#error Placeholder file.
+```
+
+See `runtime/vm/ut_j9vm.h` for an example.
```

Comment:
This step is not needed (and we don't want such a header file committed to the repository).

Reply:
I thought you had removed this section, but I see it's still here.

Reply:
It's removed now.